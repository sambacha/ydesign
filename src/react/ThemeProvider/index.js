import React from 'react';
import PropTypes from 'prop-types';
import { ThemeProvider as StyledThemeProvider } from 'styled-components';
import theme from '../theme';

const ThemeProvider = ({ children, theme, ...rest }) => {
  return (
    <StyledThemeProvider theme={theme} {...rest}>
      {children}
    </StyledThemeProvider>
  );
};

ThemeProvider.propTypes = {
  theme: PropTypes.object,
};

ThemeProvider.defaultProps = {
  theme,
};

ThemeProvider.displayName = 'ThemeProvider';

export default ThemeProvider;
