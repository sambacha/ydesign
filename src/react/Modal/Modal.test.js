import Modal from './';
import ThemeProvider from '../ThemeProvider';
import theme from '../theme';

describe('Modal component sanity', () => {
  it('has name', () => {
    expect(Modal.displayName).toBe('Modal');
  });

  it('matches default snapshot', () => {
    const component = render(<Modal />);
    expect(component).toMatchSnapshot();
  });

  it('matches themed snapshot', () => {
    const component = render(
      <ThemeProvider theme={theme}>
        <Modal />
      </ThemeProvider>
    );
    expect(component).toMatchSnapshot();
  });
});
