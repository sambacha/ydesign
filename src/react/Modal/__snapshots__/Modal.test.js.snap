// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component sanity matches default snapshot 1`] = `null`;

exports[`Modal component sanity matches themed snapshot 1`] = `null`;
